src/main/kotlin/xyz/crossward/service/GroupService.kt

```diff
@@ -67,6 +82,31 @@ class GroupService(
         groupRepository.save(savedGroup)
         return savedGroup
     }
+
+    /**
+     * Adds a User to a Group, given a GroupId and UserId
+     *
+     * @param groupId The groupId of the group to add the user to
+     * @param userID The userId of the user to add
+     * @return the created IsMember relation
+     */
+    fun addUserToGroup(groupId: String, userId: String): IsMember {
+        if (isMemberRepository.existsById(IsMemberId(userId, groupId))) {
+            throw BadRequestException("A user with ID ${userId} is already a member of a group with ID ${groupId}")
+        }
+        if (!groupRepository.existsById(groupId)) {
```

Comment:
I think you need to reverse the order of these ifs. Make sure user exists, make sure group exists, and then make sure the user isn't a member of the group.